tdb/concatenate.py

```diff
@@ -0,0 +1,36 @@
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-f', '--files', nargs='*', default=[], help="tsvs that will be concatenated")
+parser.add_argument('-o', '--output', type=str, default="data/titers_complete.tsv")
+
+def concat(files,out):
+    with open(out, 'w') as o:
+        for filename in files:
+
+            print "Concatenating and annotating %s into %s." % (filename, out)
+
+            if "cdc" in filename.lower():
+                source = "cdc"
+            elif "crick" in filename.lower():
+                source = "crick"
+            else:
+                source = "none"
+
+            if "egg" in filename.lower():
+                passage = "egg"
+            elif "cell" in filename.lower():
+                passage = "egg"
+            else:
+                passage = "none"
+
+            with open(filename, 'r') as f:
+                for line in f.readlines():
```

Comment:
Iterate through `f` instead of through the `readlines` output. The latter method loads all the lines into memory before iterating which can be wasteful while the former method uses a generator to only load batches of the input file into memory as needed.